qcodes/instrument/group_parameter.py

```diff
@@ -239,15 +239,19 @@ def _set_from_dict(self, calling_dict: Dict[str, Any]) -> None:
             raise RuntimeError("Trying to set GroupParameter not attached "
                                "to any instrument.")
         self.instrument.write(command_str)
+        for name, parameter in self.parameters.items():
+            parameter.cache.set(parameter._from_raw_value_to_value(
+                calling_dict[name]))
 
     def update(self) -> None:
         """
         Update the values of all the parameters within the group by calling
         the ``get_cmd``.
         """
         if self.instrument is None:
-            raise RuntimeError("Trying to update GroupParameter not attached "
-                               "to any instrument.")
+            raise RuntimeError("Trying to update GroupParameter not "
+                               "attached to any instrument.")
+
         ret = self.get_parser(self.instrument.ask(self.get_cmd))
         for name, p in list(self.parameters.items()):
```

Comment:
It seems to me like this logic should just be replaced by setting the cache of the parameter. But that can be done in a different pr